src/components/VideoList/VideoList.tsx

```diff
@@ -37,33 +41,43 @@ function VideoList({
   accessModel,
   isLoggedIn,
   hasSubscription,
+  hasLoadMore,
+  loadMore,
 }: Props) {
-  return (
-    <div className={classNames(styles.container, !!className && className)} data-testid={testId('video-list')}>
-      {!!header && header}
-      {playlist &&
-        playlist.playlist.map((playlistItem: PlaylistItem) => {
-          const { mediaid, title, duration, seriesId, episodeNumber, seasonNumber, shelfImage } = playlistItem;
+  const List = () => {
+    return (
+      <>
+        {playlist?.playlist?.map((playlistItem: PlaylistItem) => {
+          const { mediaid } = playlistItem;
 
           return (
             <VideoListItem
               key={mediaid}
-              title={title}
-              duration={duration}
-              image={shelfImage}
               progress={watchHistory ? watchHistory[mediaid] : undefined}
-              seriesId={seriesId}
-              episodeNumber={episodeNumber}
-              seasonNumber={seasonNumber}
               onClick={() => onListItemClick && onListItemClick(playlistItem, playlistItem.feedid)}
               onHover={() => onListItemHover && onListItemHover(playlistItem)}
               loading={isLoading}
               isActive={activeMediaId === mediaid}
               activeLabel={activeLabel}
               isLocked={isLocked(accessModel, isLoggedIn, hasSubscription, playlistItem)}
+              item={playlistItem}
             />
           );
         })}
+      </>
+    );
+  };
+
+  return (
+    <div className={classNames(styles.container, !!className && className)} data-testid={testId('video-list')}>
+      {!!header && header}
+      {loadMore ? (
+        <InfiniteScroll pageStart={0} loadMore={loadMore} hasMore={hasLoadMore} loader={<InfiniteScrollLoader key="loader" />}>
+          <List />
```

Comment:
I'd suggest to keep this inline instead of a local variable. The changes already clean up the markup quite a bit and I tend to err on the side of making the markup as WYSIWYG as possible.